tools/internal_ci/linux/grpc_e2e_performance_v2.sh

```diff
@@ -29,21 +31,48 @@ gcloud config set project grpc-testing
 gcloud container clusters get-credentials benchmarks-prod \
     --zone us-central1-b --project grpc-testing
 
+# Set up environment variables
+export PREBUILD_IMAGE_PREFIX="gcr.io/grpc-testing/e2etesting/pre_built_workers"
+export PREBUILT_IMAGE_TAG=$KOKORO_BUILD_INITIATOR-$(date '+%F-%H-%M-%S')
+export ROOT_DIRECTORY_OF_DOCKERFILES="../test-infra/containers/pre_built_workers/"
+
+# Clone test-infra repository to one upper level directory than grpc
+cd ..
+git clone --recursive https://github.com/grpc/test-infra.git
+cd grpc
+
+# Log the error and exit directly
```

Comment:
I think what you need here is to define a function to be run at exit, and then set "trap myfunction EXIT". You should keep the "set -ex" at the top.